### ISSUE TITLE
Dynamic deactivation leaves line breaks

### ISSUE THREAD

**Issue by ghost** [coderaiser/cm-show-invisibles#1]
Hi,

I am trying to use your codemirror plugin with a switch button to activate/deactivate display of invisible characters. I run into a problem which looks like a bug to me when dynamically deactivating the pluggin (set showInvisibles to false from a button for example) : spaces turn invisible again but line breaks keep showing.

Here is what I do : 
- I have a switch button that triggers the showInvisibles value of my codemirror instance.
- At the beginning, it is set to false and invisible characters are indeed not visible. 
  ![show invisible inactive](https://cloud.githubusercontent.com/assets/5527131/13169848/5dc7b95c-d6e8-11e5-8bf1-47ca66a1861d.PNG)
- I then trigger the plugin and set showInvisibles to true. Invisible characters show as expected.
  ![show invisible activated](https://cloud.githubusercontent.com/assets/5527131/13169849/5dc7e83c-d6e8-11e5-84a2-9e868ebfcb56.PNG)
- I deactivate the plugin setting showInvisibles to false again. White spaces do disapear but line breaks remain.
  ![show invisible deactivated](https://cloud.githubusercontent.com/assets/5527131/13169850/5dc8f858-d6e8-11e5-9a1a-285cc9655604.PNG)

Is this indeed a bug or did I do something wrong there ?

Best regards,
Antoine

**Comment by coderaiser** [comment]
Hi @lemairea 
This is because `cm-show-invisibles` [use css class](https://github.com/coderaiser/cm-show-invisibles/blob/70fc4de02108c382e57a6b863e59889ed5cfdbad/lib/show-invisibles.js#L107) that ads additional symbol at end of all lines. When you disable invisibles this class still exists.

Actually you can use this code as a hot fix:

``` js
var el = document.querySelector('[data-name="js-show-invisibles"]');
document.head.removeChild(el);
```

But I think there should be better solution.
Could you please tell me how you disable invisibles?

**Comment by ghost** [comment]
Hi,

Thank you for your feedback and the quick fix, I'll give it a try.

I am using code mirror within an angular directive. I put the code mirror options in my angular model and bound lineWrapping to a checkbox. Checking the checkbox sets it to true, unchecking it sets it to false without any further code.

How options are set in the codemirror directive :

``` javascript
$scope.$watch("vm.uiCodemirrorOpts", function (newValue, oldValue) {
    if (!angular.isObject(newValue)) {
        return;
    }
    angular.forEach(newValue, function (value, key) {
        if (oldValue && newValue[key] === oldValue[key]) {
            return;
        }
        mergeView.edit.setOption(key, newValue[key]);
        //Handle fullscreen
        if (key === 'fullScreen') {
            mergeView.edit.focus();
        }
    });
}, true);
```

My button :

``` html
<button type="button" 
        class="btn btn-default btn-sm btn-erlarge" 
        ng-model="fileEditor.editorOptions.showInvisibles" 
        uib-btn-checkbox >
    Show invisibles
</button>
```

Do you see any better solution ?

Thanks and best regards,
Antoine

**Comment by coderaiser** [comment]
Thank you, fixed in https://github.com/coderaiser/cm-show-invisibles/commit/1c99bed236a2ab5a721ee9ae70b84d62aa7d00ca. Landed in [v1.2.0](https://github.com/coderaiser/cm-show-invisibles/releases/tag/v1.2.0).

**Comment by coderaiser** [comment]
Landed in [v2.0.0](https://github.com/coderaiser/cm-show-invisibles/releases/tag/v2.0.0).
